server/manager/src/main/java/org/apache/accumulo/manager/upgrade/PreUpgradeValidation.java

```diff
@@ -117,4 +141,114 @@ private void fail(Exception e) {
     System.exit(1);
   }
 
+  private void validateTableLocks(final ServerContext context) {
+
+    final ZooReaderWriter zrw = context.getZooReaderWriter();
+    final ZooKeeper zk = zrw.getZooKeeper();
+    final String rootPath = context.getZooKeeperRoot();
+    final String tserverLockRoot = rootPath + Constants.ZTSERVERS;
+
+    log.debug("Looking for locks that may be from previous version in path: {}", tserverLockRoot);
+
+    AtomicInteger errorCount = new AtomicInteger(0);
+
+    List<Pair<HostAndPort,ServiceLock.ServiceLockPath>> hostsWithLocks =
+        gatherLocks(zk, tserverLockRoot);
+
+    // try a thrift call to the hosts - hosts running previous versions will fail the call
+    ThreadPoolExecutor lockCheckPool = ThreadPools.getServerThreadPools().createThreadPool(8, 32,
```

Comment:
As we discussed, I think you should just use the default ForkJoinPool here.

Reply:
Using a custom pool bounds the number of threads and keeps the global fork/join pool available - there seems to be conflicting advise on this, with some saying avoid using fork / join and others advocating for it.  I lean toward the predictability of a custom pool.